src/form-element-creator.ts

```diff
@@ -1,94 +1,35 @@
-import { ElementCreator } from "./element-creator";
+import { SelectElementCreationStrategy } from "./select-element-creation-strategy";
+import { InputCreationStrategy } from "./input-creation-strategy";
+import { FieldElementCreationStrategy } from "./field-element-creation-strategy";
 
-export class FormElementCreator {
-    private elementCreator: ElementCreator;
-
-    constructor(elementCreator: ElementCreator) {
-        this.elementCreator = elementCreator;
-    }
-
-    createInput(form: HTMLFormElement, element: any): void {
-        const input = this.elementCreator.createElement('input') as HTMLInputElement;
-        this.setInputAttributes(input, element);
-        form.appendChild(input);
-    }
-
-    createLabel(form: HTMLFormElement, element: any): void {
-        const label = this.elementCreator.createElement('label') as HTMLLabelElement;
-        label.textContent = element.for;
-        this.createInput(form, element);
-        form.appendChild(label);
-    }
-
-    createInputFields(form: HTMLFormElement, element: any): void {
-        const paragraph = this.elementCreator.createElement('div') as HTMLDivElement;
-        paragraph.textContent = element.name;
-        form.appendChild(paragraph);
-        this.createInput(form, element);
-    }
+export interface FormElementCreationStrategy {
+    create(form: HTMLFormElement, element: any): void;
+}
 
-    private setInputAttributes(input: HTMLInputElement, element: any): void {
-        input.type = element.type;
-        input.name = element.name;
-        element.class === 'buttons' ? (input.value = element.value) : (input.placeholder = element.placeholder);
-    }
+export class FormElementCreation implements FormElementCreationStrategy {
+    private selectElementCreationStrategy: SelectElementCreationStrategy;
```

Comment:
nie widze sensu tutaj. Robisz strategie i tu uzywasz konkretnych implementacji strategii jako memberow klasy. Nie czuje tego. Trzeba to przemyslec bo strategia jest od tego zeby w memberach uzywac po prostu instancji strategii a tu uzywasz konkretnej implementacji.
Predzej powinno byc private selectCreation: ElementCreationStrategy - czy tam jak kolwiek sie ta Twoja strategia nazywa